File: app/models/advisor.rb
ADVISOR_FIELD_MAPPING = {
  id: "LKB Personen-ID",
  first_name: "LKB Vorname",
  last_name: "LKB Nachname",
  nickname: "LKB Pfadiname",
  address:  "LKB Adresse",
  zip_code: "LKB PLZ",
  town:  "LKB Ort",
  country: "LKB Land",
  email: "LKB Email",
  salutation:  "LKB Anrede"
}
Advisor = Struct.new(*ADVISOR_FIELD_MAPPING.keys) do
  def self.from_csv(row)
    new(*row.to_h.slice(*ADVISOR_FIELD_MAPPING.values).values)
  end

  def equal?(other)
    id == other.id
  end
end

File: src/models/advisor.ts
import { Course } from "./course"

export type Advisor = {
  id: string;
  firstName: string;
  lastName: string;
  scoutName: string;
  address: string;
  zipcode: string;
  town: string;
  country: string;
  email: string;
  salutation: string;
}

export function advisorName(advisor: Advisor) {
  return `${advisor.firstName} ${advisor.lastName} / ${advisor.scoutName}`
}

export type AdvisorStatement = {
  advisor: Advisor;
  courses: Course[];
  year: number;
  amountPerParticipant: number;
}
